lib/router-config/src/query_planner.rs

```diff
@@ -1,10 +1,34 @@
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
-#[derive(Deserialize, Serialize, JsonSchema, Default)]
+#[derive(Debug, Deserialize, Serialize, JsonSchema, Clone)]
 pub struct QueryPlannerConfig {
     /// A flag to allow exposing the query plan in the response.
     /// When set to `true` and an incoming request has a `hive-expose-query-plan: true` header, the query plan will be exposed in the response, as part of `extensions`.
-    #[serde(default)]
+    #[serde(default = "default_query_planning_allow_expose")]
     pub allow_expose: bool,
+    /// The maximum time in milliseconds for the query planner to create an execution plan.
+    /// This acts as a safeguard against overly complex or malicious queries that could degrade server performance.
+    /// When the timeout is reached, the planning process is cancelled.
+    ///
+    /// Default: 10000 (10 seconds).
+    #[serde(default = "default_query_planning_timeout_ms")]
+    pub timeout_ms: u64,
```

Comment:
Instead of `u64`, you can use `Duration` directly here. It will allow users to easily use `1000ms` or `10s`

Reply:
and then the field name can be `timeout`